src/lib/loaders/api/loader_without_authentication_factory.ts

```diff
@@ -39,76 +39,101 @@ export const apiLoaderWithoutAuthenticationFactory = (
+                        if (apiOptions.headers) {
+                          resolve({ body, headers })
+                        } else {
+                          resolve(body)
+                        }
+                        verbose(`Requested (Uncached): ${key}`)
+                        const time = clock.end()
+                        logger(
+                          globalAPIOptions.requestIDs.requestID,
+                          apiName,
+                          key,
+                          { time, cache: false }
+                        )
+                        if (apiOptions.headers) {
+                          return cache.set(key, { body, headers })
+                        } else {
+                          return cache.set(key, body)
+                        }
+                      })
+                      .catch(err => {
+                        warn(key, err)
+                        reject(err)
+                      })
+                  }
+                )
+              } else {
                 () => {
-                  api(key, null, apiOptions)
-                    .then(({ body, headers }) => {
-                      if (apiOptions.headers) {
-                        resolve({ body, headers })
-                      } else {
-                        resolve(body)
-                      }
-                      verbose(`Requested (Uncached): ${key}`)
-                      const time = clock.end()
-                      logger(
-                        globalAPIOptions.requestIDs.requestID,
-                        apiName,
-                        key,
-                        { time, cache: false }
-                      )
-                      if (apiOptions.headers) {
-                        return cache.set(key, { body, headers })
-                      } else {
-                        return cache.set(key, body)
-                      }
-                    })
-                    .catch(err => {
-                      warn(key, err)
-                      reject(err)
-                    })
-                }
-              )
+                    api(key, null, apiOptions)
+                      .then(({ body, headers }) => {
+                        if (apiOptions.headers) {
+                          resolve({ body, headers })
+                        } else {
+                          resolve(body)
+                        }
+                        verbose(`Requested (Uncached): ${key}`)
+                        const time = clock.end()
+                        logger(
+                          globalAPIOptions.requestIDs.requestID,
+                          apiName,
+                          key,
+                          { time, cache: false }
+                        )
+                      })
+                      .catch(err => {
+                        warn(key, err)
+                        reject(err)
+                      })
```

Comment:
Can you move this into its own function and share it with the code above that handles a cache miss?

Reply:
That would be a bit DRYer but not sure it's worth it because of all the scoped variables inside the call to `api` - it's all in the context of the Promise and needs access to `api`,  `key`, `apiOptions`, `clock`, etc

Reply:
Good point.